### PR TITLE
Include runtime images into workbench images

### PR DESCRIPTION
**Pull request by harshad16**
Include runtime images into workbench images

## Description

This PR introduces runtime images which would be used as bases for Elyra pipelines 
As these are runtime images and not notebook images, they would be included directly inside the notebook images later.

Related-to: https://github.com/opendatahub-io/notebooks/issues/54
Depends-On: https://github.com/opendatahub-io/notebooks/pull/58

## How Has This Been Tested?
Build were tested local.

Runtimes image just requires to have binaries like curl, python and should allow installing packages.
Tested with following actions:
```
> cd runtimes/datascience/ubi8-python-3.8
> podman build -t runtime . --build-arg BASE_IMAGE=quay.io/opendatahub/workbench-images:base-ubi8-python-3.8-pr-63
> podman run -it localhost/runtime bash

>(app-root) (app-root) which curl
/usr/bin/curl
> (app-root) which python
/opt/app-root/bin/python
> curl https://raw.githubusercontent.com/elyra-ai/elyra/main/etc/generic/requirements-elyra.txt -o req.txt
> pip install -r req.txt
> curl https://raw.githubusercontent.com/nteract/papermill/main/papermill/tests/notebooks/simple_execute.ipynb --output simple_execute.ipynb
> python3 -m papermill simple_execute.ipynb output.ipynb
```

same check is adding in makefile for validating runtimes.

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by harshad16** [comment]
/hold

Till the dependent pr is merged.

**Comment by harshad16** [comment]
/unhold

All clear for review

File: Makefile [hunk 1]
```diff
@@ -126,6 +141,22 @@ cuda-jupyter-tensorflow-ubi9-python-3.9: cuda-jupyter-datascience-ubi9-python-3.
 jupyter-trustyai-ubi9-python-3.9: jupyter-datascience-ubi9-python-3.9
 	$(call image,$@,jupyter/trustyai/ubi9-python-3.9,$<)
 
+# Build and push runtime-datascience-ubi9-python-3.9 image to the registry
+.PHONY: runtime-datascience-ubi9-python-3.9
+runtime-datascience-ubi9-python-3.9: base-ubi9-python-3.9
+	$(call image,$@,runtimes/datascience/ubi9-python-3.9,$<)
+
+# Build and push runtime-pytorch-ubi9-python-3.9 image to the registry
+.PHONY: runtime-cuda-pytorch-ubi9-python-3.9
```
Comment (atheo89) [inline Makefile:150]: The PyTorch notebook is derived from the data science, not from CUDA. So, as base image should have `runtime-datascience-runtime-datascience-ubi9-python-3.9` and the name shall be like: runtime-pytorch-ubi9-python-3.9.

File: runtimes/datascience/ubi9-python-3.9/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,25 @@
+FROM quay.io/opendatahub/workbench-images:runtime-minimal-ubi9-python-3.9-pr-58
```
Comment (atheo89) [inline runtimes/datascience/ubi9-python-3.9/Dockerfile:1]: This should be replaced by ARG, or it needs to rebase the PR

**Comment by atheo89** [comment]
Furthermore, could you add instructions on the PR description on how you can test these changes?

File: runtimes/datascience/ubi9-python-3.9/Dockerfile (hunk 2 above)
Comment (harshad16) [reply runtimes/datascience/ubi9-python-3.9/Dockerfile]: Good catch
fixed it.

File: Makefile [hunk 3]
```diff
@@ -79,6 +79,21 @@ cuda-jupyter-datascience-ubi8-python-3.8: cuda-jupyter-minimal-ubi8-python-3.8
 cuda-jupyter-tensorflow-ubi8-python-3.8: cuda-jupyter-datascience-ubi8-python-3.8
 	$(call image,$@,jupyter/tensorflow/ubi8-python-3.8,$<)
 
+# Build and push runtime-datascience-ubi8-python-3.8 image to the registry
+.PHONY: runtime-datascience-ubi8-python-3.8
+runtime-datascience-ubi8-python-3.8: base-ubi8-python-3.8
+	$(call image,$@,runtimes/datascience/ubi8-python-3.8,$<)
+
+# Build and push runtime-pytorch-ubi8-python-3.8 image to the registry
+.PHONY: runtime-cuda-pytorch-ubi8-python-3.8
```
[the remark this answers is not in the document]
Comment (harshad16) [reply Makefile:88]: Ack, runtime images are not like notebook images, 
we had a chained build of minimal -> data-science -> pytorch/tf , as we had scripts in minimal and data science which needed to flow downwards.

in runtime, we just build the image and each image doesn't need to be directly dependent.

File: Makefile (hunk 1 above)
Comment (harshad16) [reply Makefile]: as explained in above comment, runtime image just need to be based out of base image.
Doesnt need to dependent as of now.

File: Makefile [hunk 4]
```diff
@@ -126,6 +141,22 @@ cuda-jupyter-tensorflow-ubi9-python-3.9: cuda-jupyter-datascience-ubi9-python-3.
...
+runtime-datascience-ubi9-python-3.9: base-ubi9-python-3.9
+	$(call image,$@,runtimes/datascience/ubi9-python-3.9,$<)
+
+# Build and push runtime-pytorch-ubi9-python-3.9 image to the registry
+.PHONY: runtime-cuda-pytorch-ubi9-python-3.9
+runtime-cuda-pytorch-ubi9-python-3.9: cuda-ubi9-python-3.9
+	$(call image,$@,runtimes/pytorch/ubi9-python-3.9,$<)
+
+# Build and push runtime-cuda-tensorflow-ubi9-python-3.9 image to the registry
+.PHONY: runtime-cuda-tensorflow-ubi9-python-3.9
```
[the remark this answers is not in the document]
Comment (harshad16) [reply Makefile:155]: as explained in above comment, runtime image just need to be based out of base image.
Doesnt need to dependent as of now.

**Comment by harshad16** [comment]
Ready for review
updated all the requested changes.

**Comment by atheo89** [comment]
Thank you, @harshad16! 
Typically, when we add a new image to the notebooks repository, we need to first merge the pull request from the [release](https://github.com/openshift/release/pull/38165) repository. This allows us to test the new images and verify their functionality. Therefore, I suggest that we change the status of PR https://github.com/openshift/release/pull/38165 from draft to open. This will enable us to run tests on the new runtime images and ensure that everything is working properly before merging the PR. 

Overall, /lgtm

**Review by VaishnaviHire — COMMENTED** [review]
/lgtm

**Pull request merged**